src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -1525,9 +1519,9 @@ protected internal override void OnProjectRemoved(ProjectId projectId)
 
                 base.OnProjectRemoved(projectId);
 
-                _deferredJoinableTasks.Add(_threadingContext.JoinableTaskFactory.RunAsync(async () =>
+                _threadingContext.ShutdownBlockingTasks.Add(_threadingContext.JoinableTaskFactory.RunAsync(async () =>
```

Comment:
we could also consider an `AddShutdownBlockWork(Action<CT>)` method that is passed a lambda which is runs, passing in teh _threadingContext disposal token.  that way we don't risk the code using the wrong token.